Content.Server/GameTicking/Rules/RandomRuleSystem.cs

```diff
@@ -0,0 +1,61 @@
+using Content.Server.Chat.Managers;
+using Content.Server.GameTicking.Rules.Components;
+using Content.Server.StationEvents;
+using Content.Shared.GameTicking.Components;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Content.Shared.Storage;
+using System.Linq;
+
+namespace Content.Server.GameTicking.Rules;
+
+
+/// <summary>
+/// Entities spawned with this component will try to add gamerules from the provided EntitySpawnEntry list.
+/// This can be used to subset gamerules for balance, or to do something else like add it to a prototype if you want ninjas to start with meteors etc.
+/// </summary>
+public sealed class RandomRuleSystem : GameRuleSystem<RandomRuleComponent>
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IChatManager _chatManager = default!;
+    [Dependency] private readonly EventManagerSystem _event = default!;
+
+    protected override void Started(EntityUid uid, RandomRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
+    {
+        var selectedRules = EntitySpawnCollection.GetSpawns(component.SelectableGameRules, _random);
+
+        int ruleQuant = component.MinMaxRules.Next(_random);
+
+        if (selectedRules == null || selectedRules.Count == 0)
+            return;
+
+        _random.Shuffle(selectedRules);
+        var nRules = selectedRules.Take(ruleQuant); // Does not allow duplicate selection, unless the EntitySpawnCollection had duplicate entries.
+
+        var availableEvents = _event.AvailableEvents(); // handles the player counts and individual event restrictions, we need to do it each i incase it changes
+
+        foreach (var rule in nRules)
+        {
+            var ruleEnt = new EntityUid();
+
+            if (!_prototypeManager.TryIndex(rule, out var ruleProto))
+            {
+                Log.Warning("The selected random rule is missing a prototype!");
+                continue;
+            }
+
+            if (!availableEvents.ContainsKey(ruleProto))
+            {
+                Log.Warning("The selected random rule is not available!");
+                continue;
+            }
+            GameTicker.StartGameRule(rule, out ruleEnt);
+
+            var str = Loc.GetString("station-event-system-run-event", ("eventName", ToPrettyString(ruleEnt)));
+            _chatManager.SendAdminAlert(str);
+            Log.Info(str);
+        }
```

Comment:
Isn't this already admin logged, hardcoding in logs for 1 rule looks like a workaround.

Reply:
Fixed it, but it does do this at round init now, and I don't see an easy way around that without special casing again.
![image](https://github.com/user-attachments/assets/c4726a31-76c9-4a7e-8f41-6f9086256743)

Reply:
Maybe I could suppress it before round start? 🤔